tests/Acoustics.Test/AudioAnalysisTools/Events/EventFilterTests.cs

```diff
@@ -0,0 +1,129 @@
+// <copyright file="EventFilterTests.cs" company="QutEcoacoustics">
+// All code in this file and all associated files are the copyright and property of the QUT Ecoacoustics Research Group (formerly MQUTeR, and formerly QUT Bioacoustics Research Group).
+// </copyright>
+
+namespace Acoustics.Test.AudioAnalysisTools.Events
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Text;
+    using global::AudioAnalysisTools.Events;
+    using global::AudioAnalysisTools.Events.Types;
+    using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+    [TestClass]
+    public class EventFilterTests
+    {
+        [TestMethod]
+        public void TestRemoveEnclosedEvent()
+        {
+            var ev1 = new SpectralEvent()
+            {
+                Name = "ev1",
+                EventEndSeconds = 9,
+                EventStartSeconds = 1,
+                HighFrequencyHertz = 300,
+                LowFrequencyHertz = 200,
+            };
+
+            var ev2 = new SpectralEvent()
+            {
+                Name = "ev2",
+                EventEndSeconds = 9,
+                EventStartSeconds = 1,
+                HighFrequencyHertz = 500,
+                LowFrequencyHertz = 400,
+            };
+
+            var ev3 = new SpectralEvent()
+            {
+                Name = "ev3",
+                EventEndSeconds = 9,
+                EventStartSeconds = 1,
+                HighFrequencyHertz = 500,
+                LowFrequencyHertz = 400,
+            };
+
+            // test with combination of events
+            var listOfEvents = new List<EventCommon>() { ev1, ev2, ev3 };
+            var events = CompositeEvent.RemoveEnclosedEvents(listOfEvents);
+
+            Assert.AreEqual(2, events.Count);
+            var @event = (SpectralEvent)events[0];
+
+            Assert.AreEqual(1, @event.EventStartSeconds, 0.1);
+            Assert.AreEqual(9, @event.EventEndSeconds, 0.1);
+            Assert.AreEqual(200, @event.LowFrequencyHertz, 0.1);
+            Assert.AreEqual(300, @event.HighFrequencyHertz, 0.1);
+            Assert.AreEqual("ev1", @event.Name);
+
+            @event = (SpectralEvent)events[1];
+            Assert.AreEqual(1, @event.EventStartSeconds, 0.1);
+            Assert.AreEqual(9, @event.EventEndSeconds, 0.1);
+            Assert.AreEqual(400, @event.LowFrequencyHertz, 0.1);
+            Assert.AreEqual(500, @event.HighFrequencyHertz, 0.1);
+            Assert.AreEqual("ev2", @event.Name);
```

Comment:
Unless I am misreading this, it could be simplified to:

```suggestion
            CollectionAssert.AreEqual(new { ev1, ev2 }, events)
```

Reply:
And this would be another case where delta thresholds are not appropriate